plugins/home/src/homePageComponents/StarredEntities/Content.tsx

```diff
@@ -32,29 +34,66 @@ import {
 } from '@material-ui/core';
 import StarIcon from '@material-ui/icons/Star';
 import React from 'react';
+import useAsync from 'react-use/lib/useAsync';
 
 /**
  * A component to display a list of starred entities for the user.
  *
  * @public
  */
 export const Content = () => {
+  const catalogApi = useApi(catalogApiRef);
   const catalogEntityRoute = useRouteRef(entityRouteRef);
   const { starredEntities, toggleStarredEntity } = useStarredEntities();
 
+  // Grab starred entities from catalog to ensure they still exist and also retrieve display titles
+  const entities = useAsync(async () => {
+    if (!starredEntities.size) {
+      return [];
+    }
+
+    const filter = [...starredEntities]
+      .map(ent => parseEntityRef(ent))
+      .map(ref => ({
+        kind: ref.kind,
+        'metadata.namespace': ref.namespace,
+        'metadata.name': ref.name,
+      }));
+
+    return (
+      await catalogApi.getEntities({
+        filter,
+        fields: [
+          'kind',
+          'metadata.namespace',
+          'metadata.name',
+          'metadata.title',
+        ],
+      })
+    ).items;
+  }, [catalogApi, starredEntities]);
+
   if (starredEntities.size === 0)
     return (
       <Typography variant="body1">
         You do not have any starred entities yet!
       </Typography>
     );
 
-  return (
+  if (entities.loading) {
+    return <Progress />;
+  }
+
+  return entities.error ? (
+    <Typography>Error loading entities: {entities.error}</Typography>
```

Comment:
```
if (error) {
    return <ResponseErrorPanel error={error} />;
}
```

this is the classic by now! Dedicated for the purpose, has more debugging info available etc

Reply:
Thanks for the tip! Updated!